pkgs/development/rocm-modules/6/default.nix

```diff
@@ -1,530 +1,815 @@
+    };
+
+    rocwmma = callPackage ./rocwmma {
+      inherit
+        rocmUpdateScript
+        rocm-cmake
+        rocm-smi
+        rocblas
+        clr
+        ;
+      inherit (llvm) openmp;
+      stdenv = llvm.rocmClangStdenv;
+    };
+
+    rocalution = callPackage ./rocalution {
+      inherit
+        rocmUpdateScript
+        rocm-cmake
+        rocprim
         rocsparse
-        rccl
-        rocthrust
+        rocrand
         rocblas
-        hipsparse
-        hipfort
-        rocwmma
-        hipsolver
-        rocsolver
-        rocm-hip-libraries
-        rocm-hip-runtime-devel
-      ];
+        clr
+        ;
+      inherit (llvm) openmp;
+      stdenv = llvm.rocmClangStdenv;
+    };
+
+    rocmlir = callPackage ./rocmlir {
+      inherit
+        rocmUpdateScript
+        rocm-cmake
+        rocminfo
+        clr
+        ;
+      stdenv = llvm.rocmClangStdenv;
+      buildRockCompiler = true;
     };
 
-    rocm-hip-libraries = symlinkJoin {
-      name = "rocm-hip-libraries-meta";
+    rocmlir-rock = rocmlir;
+    # rocmlir-rock = rocmlir.override {
+    #   buildRockCompiler = true;
+    # };
 
-      paths = [
+    hipsolver = callPackage ./hipsolver {
+      inherit
+        rocmUpdateScript
+        rocm-cmake
         rocblas
-        hipfort
-        rocm-core
         rocsolver
-        rocalution
-        rocrand
-        hipblas
-        rocfft
-        hipfft
-        rccl
-        rocsparse
-        hipsparse
-        hipsolver
-        rocm-hip-runtime
-      ];
+        clr
+        ;
+      stdenv = llvm.rocmClangStdenv;
     };
 
-    rocm-openmp-sdk = symlinkJoin {
-      name = "rocm-openmp-sdk-meta";
+    hipblas-common = callPackage ./hipblas-common {
+      inherit rocm-cmake;
+      stdenv = llvm.rocmClangStdenv;
+    };
 
-      paths = [
-        rocm-core
-        llvm.clang
-        llvm.mlir
-        llvm.openmp # openmp-extras-devel (https://github.com/ROCm/aomp)
-        rocm-language-runtime
-      ];
+    hipblas = callPackage ./hipblas {
+      inherit
+        rocmUpdateScript
+        rocm-cmake
+        rocblas
+        rocsolver
+        clr
+        ;
+      stdenv = llvm.rocmClangStdenv;
+    };
+
+    hipblaslt = callPackage ./hipblaslt {
+      inherit rocmUpdateScript;
+      inherit (llvm) openmp clang-sysrooted;
+      stdenv = llvm.rocmClangStdenv;
     };
 
-    rocm-opencl-sdk = symlinkJoin {
-      name = "rocm-opencl-sdk-meta";
+    # hipTensor - Only supports GFX9
 
-      paths = [
-        rocm-core
-        rocm-runtime
-        clr
-        clr.icd
-        rocm-thunk
-        rocm-opencl-runtime
-      ];
+    composable_kernel_build = callPackage ./composable_kernel {
+      inherit rocmUpdateScript rocm-cmake clr;
+      inherit (llvm) rocm-merged-llvm;
+      stdenv = llvm.rocmClangStdenv;
     };
 
-    rocm-opencl-runtime = symlinkJoin {
-      name = "rocm-opencl-runtime-meta";
+    # FIXME: we have compressed code objects now, may be able to skip two stages?
```

Comment:
CCOB is on by default for composable_kernel for ROCm v6.2+, so yes no need for two-stage builds now.